tests/infra.py

```diff
@@ -15,6 +17,17 @@ def start_verbose_logging():
             logging.getLogger(logger_name).setLevel(logging.WARNING)
 
 
+def progress(message):
+    """
+    Print progress commentary if env 'VERBOSE' is set to anything.
+
+    Note it doesn't include a newline so you can produce dots... or "doing...done" lines.
+    """
+    if 'VERBOSE' in os.environ:
```

Comment:
i would just quickly peruse https://docs.python.org/3/library/logging.html to see if there's anything that's appropriate there.